File: mobile/lib/features/categories/models/category.dart
class Category {
  final String id;
  final String name;
  final String description;

  const Category({required this.id, required this.name, this.description = ''});

  factory Category.fromJson(Map<String, dynamic> json) => Category(
        id: (json['id'] ?? json['_id']).toString(),
        name: json['name']?.toString() ?? '',
        description: json['description']?.toString() ?? '',
      );

  Map<String, dynamic> toJson() => {'name': name, 'description': description};
}

File: backend/src/models/Category.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true, unique: true },
    description: { type: String, default: '' },
    isDeleted: { type: Boolean, default: false },
    deletedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User', default: null },
    deletedAt: { type: Date, default: null },
    deleteReason: { type: String, default: '' },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Category', categorySchema);
